File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    
    answer = []
    
    for command in commands:
        i, j, k = command[0], command[1], command[2]
        split_array = array[i-1:j]
        split_array.sort()
        answer.append(split_array[k-1])
        
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    
    vector<int> answer;
    
    for (auto c: commands) {
        int i = c[0];
        int j = c[1];
        int k = c[2];
        
        vector<int> sub(array.begin()+(i-1), array.begin()+j);
        sort(sub.begin(), sub.end());
        answer.push_back(sub[k-1]);
    }
    
    return answer;
}